### PR TITLE
Upgrade nx

### PR DESCRIPTION
**Pull request by georg-schwarz**
Upgraded NX to latest version and fixed emerging issues

- NX migration via `nx migrate latest` and `nx migrate --run-migrations`. I didn't check in the `migrations.json` file
- Linter seems to have slightly changed,  so I fixed it all required places
- Fixed webpack configurations (missing config file not added by migration)
- Remove conflicting linter rules from `monaco-editor`. The optimal way would be to remove the conflict, but we don't write much react anyway, so I'd just remove the jvalue eslint as a workaround...

## Open Issues

- the docusaurus plugin for nx is outdated and still references old `@nrwl` packages - not fixable right now

### CODE REVIEW

File: libs/language-server/project.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
       "options": {
         "commands": [
           "langium generate -f libs/language-server/langium-config.json",
-          // Workaround until https://github.com/langium/langium/issues/740 is resolved:
           "node tools/scripts/fix-monarch-grammar-escape.mjs",
```
Comment (joluj) [inline libs/language-server/project.json:12]: Why is this comment gone? It's not resolved is it?

File: jest.preset.js [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+  ...nxPreset,
+  /* TODO: Update to latest Jest snapshotFormat
+   * By default Nx has kept the older style of Jest Snapshot formats
+   * to prevent breaking of any existing tests with snapshots.
+   * It's recommend you update to the latest format.
+   * You can do this by removing snapshotFormat property
+   * and running tests with --update-snapshot flag.
+   * Example: "nx affected --targets=test --update-snapshot"
+   * More info: https://jestjs.io/docs/upgrading-to-jest29#snapshot-format
+   */
```
Comment (joluj) [inline jest.preset.js:17]: Do we use snapshots? If not we can remove this comment.

File: apps/vs-code-extension/project.json [hunk 3]
```diff
@@ -38,28 +43,22 @@
...
       }
     },
     "pack": {
       "executor": "nx:run-commands",
       "dependsOn": ["build"],
       "options": {
         "commands": [
-          // Delete vscode dependency in package.json which was generated by Nx
           "node tools/scripts/delete-dependencies.mjs vs-code-extension",
-          // Add an empty license file, so vsce does not complain during packaging
```
Comment (joluj) [inline apps/vs-code-extension/project.json:62]: Keep comments

File: apps/vs-code-extension/project.json [hunk 4]
```diff
@@ -17,13 +17,18 @@
...
+            "output": "/"
+          }
+        ],
+        "generatePackageJson": true,
+        "thirdParty": true
       },
       "configurations": {
         "dev": {
           "esbuildOptions": {
-            // Required, so VS Code's debugger is able to attach when running the extension
```
Comment (joluj) [inline apps/vs-code-extension/project.json:26]: Keep comment

File: apps/language-server-web-worker/project.json [hunk 5]
```diff
@@ -36,26 +35,21 @@
...
-        "passWithNoTests": true
+        "jestConfig": "apps/language-server-web-worker/jest.config.ts"
       }
     },
     "publish": {
       "executor": "nx:run-commands",
       "dependsOn": ["build"],
       "options": {
         "commands": [
-          // Manually delete all dependencies in package.json because the "externalDependencies": "none" setting in "build" above seems to have no effect on the generated package.json
```
Comment (joluj) [inline apps/language-server-web-worker/project.json:58]: Keep

**Review by joluj — COMMENTED** [review]
There are some comments I'd rather still have included. 

NX supports reading JSON files with comments but they get stripped when updating (See https://github.com/nrwl/nx/issues/4244#issuecomment-743329379)

There is also a [workaround](https://github.com/nrwl/nx/issues/9983#issuecomment-1331372285), I'm not sure if I like it though :D

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -39,9 +39,6 @@ jobs:
       - name: Build
         run: npx nx affected --target=build --configuration=prod --parallel=3
 
-      - name: Lint Nx specific workspace files
-        run: npx nx workspace-lint
```
Comment (georg-schwarz) [inline .github/workflows/ci.yaml:43]: FIY - `workspace-lint` was removed with version 15 of nx

**Review by joluj — APPROVED** [review]

**Pull request merged**